code/game/objects/machinery/floodlight.dm

```diff
@@ -136,6 +136,228 @@
 /obj/machinery/floodlight/landing/attackby()
 	return
 
+/obj/machinery/floodlightcombat
+	name = "Armoured floodlight"
+	icon = 'icons/obj/machines/floodlight.dmi'
+	icon_state = "floodlightcombat_off"
+	anchored = FALSE
+	density = TRUE
+	/// Determines how much light does the floodlight make , every light tube adds 4 tiles distance.
+	var/brightness = 0
+	/// Turned ON or OFF? Used for the overlays and power usage.
+	var/on = 0
+	/// Used to show if the object is tipped
+	var/tipped
+	resistance_flags = UNACIDABLE
+	appearance_flags = KEEP_TOGETHER
+	idle_power_usage = 50
+	active_power_usage = 2500
+	wrenchable = TRUE
+
+/// Handles the wrench act .
+/obj/machinery/floodlightcombat/wrench_act(mob/living/user, obj/item/I)
+	. = ..()
+	to_chat(user , "<span class='notice'>You begin wrenching [src]'s bolts")
+	playsound( loc, 'sound/items/ratchet.ogg', 60, FALSE)
+	if(!do_after(user, 2 SECONDS, TRUE, src))
+		return FALSE
+	if(anchored)
+		to_chat(user , "<span class='notice'>You unwrench [src]'s bolts")
+		anchored = FALSE
+	else
+		to_chat(user , "<span class='notice'>You wrench down [src]'s bolts")
+		anchored = TRUE
+	set_light(0, 5, "#C5E3E132")
+	on = 0
+
+/// Visually shows that the floodlight has been tipped and breaks all the lights in it.
+/obj/machinery/floodlightcombat/proc/tip_over()
+	for(var/obj/item/light_bulb/tube/T in contents)
+		T.status = 2
```

Comment:
Are there no defines for this?

Reply:
There is a define but whenever i try to use it here it says its undefined.

Reply:
Move the define to the _DEFINES folder then